### PR TITLE
[JW8-11314] Keyboard shortcuts toggle improvements.

### PR DESCRIPTION
**Pull request by zetagame**
### This PR will...
- Improve accessibility functionality of keyboard shortcuts toggle switch
- Fix vertical centering issues with switch
- Fix strange outline on focus
- Improve type-checks on keyboard shortcuts toggle handlers

### Why is this Pull Request needed?
The functionality of the button was vague when using a screen-reader.

#### Addresses Issue(s):

JW8-11314

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

File: src/js/view/controls/shortcuts-tooltip.ts [hunk 1]
```diff
@@ -116,14 +116,18 @@ export default function (
     };
 
     const documentClickHandler = (e: Event) => {
-        const target = e.target as HTMLElement;
-        if (!/jw-shortcuts|jw-switch/.test(target.className)) {
+        const target: EventTarget | null = e.target;
+        if (target instanceof HTMLElement && !/jw-shortcuts|jw-switch/.test(target.className)) {
```
Comment (robwalch) [inline src/js/view/controls/shortcuts-tooltip.ts:120]: Can we use `!== null` or simply `target` here?

File: src/js/view/controls/shortcuts-tooltip.ts [hunk 2]
```diff
@@ -116,14 +116,18 @@ export default function (
...
             close();
         }
     };
 
     const toggleClickHandler = (e: Event) => {
-        const toggle = e.currentTarget as HTMLElement;
+        if (!(e.currentTarget instanceof HTMLElement)) {
```
Comment (robwalch) [inline src/js/view/controls/shortcuts-tooltip.ts:126]: I'd really rather not use runtime `instanceof ` checks. Why the change?

File: src/js/view/controls/shortcuts-tooltip.ts (hunk 1 above)
Comment (robwalch) [reply src/js/view/controls/shortcuts-tooltip.ts:120]: If `Event` `target` and `currentTarget` are either an element or null, it's much cheaper to just test that it's not `null` or truthy. We don't have a dependency on `HTMLElement` as global, or minor performance impact of looking up the prototype chain.

File: src/js/view/controls/shortcuts-tooltip.ts (hunk 2 above)
Comment (zetagame) [reply src/js/view/controls/shortcuts-tooltip.ts:126]: as HTMLElement is basically saying "Ignore the check saying this isn't a typesafe operation, no matter what it's an HTMLElement". Since it's based on the target, we would need to check at runtime to know.

Comment (robwalch) [reply src/js/view/controls/shortcuts-tooltip.ts:126]: given that we added the listener to an element, why do we need the check? We didn't need it before TypeScipt, do adding runtime checks to satisfy a build time library is a de-optimization.

File: src/js/view/controls/shortcuts-tooltip.ts (hunk 1 above)
Comment (zetagame) [reply src/js/view/controls/shortcuts-tooltip.ts:120]: `e.target` is an `EventTarget` when defined, which isn't necessarily an HTMLElement. While instanceof does incur about a 1/24000 seconds cost to check, this is comparable to checking if it has classname and ensures that this is a type-safe function. That being said, if we don't mind reading e.target twice there's a cleaner way to write it. See latest commit @robwalch .

File: src/js/view/controls/shortcuts-tooltip.ts (hunk 2 above)
Comment (zetagame) [reply src/js/view/controls/shortcuts-tooltip.ts:131]: The casting is preventing a compile time error.

Comment (robwalch) [reply src/js/view/controls/shortcuts-tooltip.ts:131]: You could also check `e.currentTarget.nodeType === 1`, but either way it doesn't relate directly to what this method is meant to do which is to change the `aria-checked` attribute value on the element you attached the listener to which is `shortcutToggleUi`, which should also be the value of `this`.

Comment (zetagame) [reply src/js/view/controls/shortcuts-tooltip.ts]: I'm worried about the edge cases, especially in the case of documentClickHandler. `e.currentTarget` might be an element at the time that it's clicked, but removed from the DOM by the time the handler fires. That being said, I do understand not wanting to add work beyond the ticket scope so I've reverted the type-check enhancements.

Comment (robwalch) [reply src/js/view/controls/shortcuts-tooltip.ts]: You can set attributes on an element no longer in the DOM. If the point it to modify the element the listener is attached to, then just use `this` instead of `event.currentTarget`.

**Review by robwalch — APPROVED** [review]
